### PR TITLE
Refactor whatif promise

### PR DESCRIPTION
**Pull request by reneolivo**
The WhatIfPromise has been revamped from the ground up.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,31 +1,26 @@
...
-**Executed functions:**
-```
-whatIf(theUserIs('admin'))
-.then(displayAdminMenu);
-```
-
-**Logical operators:**
+**You can pass logical operators:**
 ```
 whatif(theThemeColor === 'red')
```
Comment (rmariuzzo) [inline README.md]: Should be `whatIf`.

File: spec/whatif.spec.js [hunk 2]
```diff
@@ -4,121 +4,21 @@ let whatIf = require('../src/whatif');
 let WhatIfPromise = require('../src/promise');
 
 describe('WhatIf', () => {
-  it('should be defined', () => {
-    expect(typeof whatIf).toBe('function');
+  let whatIfSpy;
+
+  beforeEach(() => {
+    const wrapper = { WhatIfPromise: WhatIfPromise };
```
Comment (rmariuzzo) [inline spec/whatif.spec.js]: You could use **object literal property value shorthand** as: `{ WhatIfPromise }`

**Review by rmariuzzo — CHANGES_REQUESTED** [review]
@reneolivo I added a few comment that I consider should be addressed.

File: spec/whatif.spec.js (hunk 2 above)
Comment (reneolivo) [reply spec/whatif.spec.js]: I completely forgot about that... thanks!!!

Comment (rmariuzzo) [reply spec/whatif.spec.js]: You are welcome sir 🙇

**Comment by reneolivo** [comment]
Thanks for the tips! I'll be merging this one now.

**Pull request merged**